Support wildcard character (*) for looking up dependencies.

pferrot opened the pull request.

Commit:
Support wildcard character (*) for looking up dependencies.

opaugam commented:
good stuff patrice - i really like the fact you are matching the coding style nicely :)

Q : did you test it out in a real-life use-case ? i'm curious to see if there is any potential performance issue ? looks good in any case.

Q2 : do we need a full blown regex match or can/should we use the trimmed-down fnmatch package ?

pferrot commented:
I just tested in dev with 4 or 5 clusters. Adding/removing/scaling up or down clusters worked smoothly.

Regarding regex VS fnmatch: both will work, up to you :-)

opaugam commented:
awesome - i'll perform a few tests & merge into 1.0.6 later on today !

Commit:
Build Docker image from pferrot/ochopod (and not from the original
autodesk-cloud/ochopod)

Commit:
Provide "depends_on" information when fetching pod info.

Commit:
Rollback previous commit.

Commit:
Merged from upstream.

Commit:
Revert unnecessary change.

Commit:
Fixed bug where _1() does not make an exception but returns an empty
'hint', hence failing to find where Zookeeper is located since not
proceeding to _2() or _3().

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Rolled back temp fix since root cause fixed in upstream.

Commit:
Removed new line as in upstream.

opaugam commented:
yes i'm going to finally include this one :)

opaugam commented on ochopod/watchers/remote.py:
```diff
@@ -80,12 +81,39 @@ def spin(self, data):
+                            if self.zk.exists(fullPath, watch=self.feedback):                            
+                                try:
+                                    value, stat = self.zk.get(fullPath, watch=self.feedback)
+                                    pods = {k: v for (k, v) in (json.loads(value).items() + pods.items())}
```
i believe the pods dict will get overriden in that for loop .. shouldn't it be a concatenation instead of an assignment ?

pferrot replied:
I believe it is fine thanks to the +

```
json.loads(value).items() + pods.items()
```

Maybe it could be more elegant...

opaugam replied:
oh sorry - that escaped my attention / you are correct

opaugam replied:
did i tell you you did a great job btw ? that was not super trivial..

opaugam commented:
i'm going to merge it manually along with other changes / closing this PR (push coming soon..)

pferrot commented:
Awesome, thanks! And I like the "absolute" dependency feature - I am sure that I will use it at some point!
